### ISSUE TITLE
User preference for reduced focus visibility

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7790]
### Description
Add a setting in accessibility preferences for reduced focus visibility. Will switch focus styles from displaying on `:focus` to `:focus-visible` for those who opt into reduced focus visibility (e.g. mouse users who prefer not to see focus outlines when clicking elements into focus).

**Comment by michtio** [comment]
I think it's smarter to do it the other way around, use `:focus-visible` as a standard, as there will be much more users not needing the added accessibility rather than the ones needing it ( on the CP level ). 

So I'd rather see "opt-in for enhanced focus visibility".

Just my 2 cents.

**Comment by mmikkel** [comment]
In general I think Craft should err on the side of accessibility and make any accessibility related user preference opt-out. Either way, I assume this new preference will be added to the [`accessibilityDefaults`](https://craftcms.com/docs/3.x/config/config-settings.html#accessibilitydefaults) config setting, so it'll be easy enough for developers to decide if it should in fact be enabled or disabled by default.

**Comment by michtio** [comment]
well @mmikkel I can see why people might think differently, but if I boot up my Mac , the accessibility features are also turned off by default...

Kinda saying "new computer, have everything enabled" eg. VoiceOver, Zoom, ... because most accessibility related user preferences should be opt-out. 

We can keep on discussing this of course and everyone will have their own opinion about it, so it would be never ending, just wanted to give my point of view. 

As for argument I find the client/developer should determine how much accessibility features the end product needs.

**Comment by mmikkel** [comment]
@michtio Well, I don't think VoiceOver compares to something like a focus outline (or differently shaped element status icons). It's a pretty subtle/non intrusive change to the UI :)

But yeah, let me rephrase – I think any accessibility-related user preference that doesn't completely change the appearance or function of the UI (like, for example, inverted or grayscale modes would, if those were a thing) or is overly intrusive (like something like VoiceOver would be, to a person who doesn't need it), should be opt-out. But again, as long as these preferences are added to the `accessibilityDefaults`, the developer will get to decide – so whatever Craft's default is might not matter all that much.

**Comment by missmatsuko** [comment]
The plan is to have accessibility settings enabled by default, which is aligned with [ATAG 2.0 B.4.1.1 Features Active by Default: All accessible content support features are turned on by default](https://www.w3.org/TR/2015/NOTE-IMPLEMENTING-ATAG20-20150924/#sc_b411). 

There might be some situations where having an accessibility feature on will actually interfere with accessibility for some individuals (e.g. VoiceOver on Mac could make the OS unusable for some users), but I think a focus outline is at most an annoyance for those who don't need it.

**Comment by brandonkelly** [comment]
> well @mmikkel I can see why people might think differently, but if I boot up my Mac , the accessibility features are also turned off by default...

macOS actually gives users the ability to opt into most accessibility settings right on setup (see https://twitter.com/brandonkelly/status/1355732411519266823?s=21). And if you stare at the screen long enough, VoiceOver kicks in automatically.

**Comment by mmikkel** [comment]
@brandonkelly It could actually be a really good idea if Craft had a similar onboarding experience for new users, the first time they logged in to the CP.

**Comment by brandonkelly** [comment]
Agree :)

**Comment by khalwat** [comment]
We had a parallel discussion here @brandonkelly -- made an issue about it https://github.com/craftcms/cms/issues/7809

But we reached the same conclusion I think :)

**Comment by jsunsawyer** [comment]
Testing out 3.7 beta now. Not seeing this setting have any effect at the moment, but wanted to point out another thing that might be nice to address sooner rather than later.

![Screen Shot 2021-06-18 at 1 31 48 PM](https://user-images.githubusercontent.com/1007191/122613502-cc0c3f00-d039-11eb-9ba2-3cb00585ad6d.png)

These user account preferences for accessibility are inconsistent/confusing. Most accessibility options seem to be set up so that checking a setting makes the site more accessible. The `Reduce focus visibility` setting is the opposite, and might be unclear/confusing for users. One would think that checking all these settings would make the site the most accessible.

It might be more clear to have the setting be `Increase focus visibility` (or similar) and have this be checked by default, as it seems this is the desired initial state for this preference.

**Comment by khalwat** [comment]
@jsunsawyer  wrote:
> These user account preferences for accessibility are inconsistent/confusing. Most accessibility options seem to be set up so that checking a setting makes the site more accessible. The Reduce focus visibility setting is the opposite, and might be unclear/confusing for users. One would think that checking all these settings would make the site the most accessible.

Agree. Maybe it should be phrased "Use focus rings" or the like, off by default?

**Comment by mmikkel** [comment]
It would definitely be more intuitive if all of the accessibility settings were either opt-in or opt-out, even if that would mean that some of them would be checked by default, and others not (although IMO all of the current ones could just as well be enabled by default).

**Comment by khalwat** [comment]
@mmikkel or just let people choose the first time they log in, as Apple does with its accessibility settings per: https://github.com/craftcms/cms/issues/7790

**Comment by mmikkel** [comment]
@khalwat Absolutely, that'd be be the ideal way to handle this, but in lieu of that – I assume that an interactive onboarding flow is not something we're going to see for some time yet.

**Comment by brandonkelly** [comment]
Just changed the preference to “Always show focus rings” for the next beta (aabbab2233cded93c29acfbf08c0990e4b050b41).

> Not seeing this setting have any effect at the moment

@jsunsawyer It uses the `:focus-visible` pseudo class, which [isn’t yet supported by Safari](https://caniuse.com/?search=focus-visible). Maybe that’s why?

**Comment by jsunsawyer** [comment]
@brandonkelly I'm using the latest version of Chrome on macOS.

I think that it might be applying, but just isn't consistent. Here's an example of content tabs in a globals section maintaining an outline after selecting a tab.

![Screen Shot 2021-06-20 at 9 32 32 AM](https://user-images.githubusercontent.com/1007191/122681726-9fc40000-d1aa-11eb-98bf-9a9404cceec5.png)

(This is in beta.1 – as I'm testing from the hosted Craft demo)

**Comment by brandonkelly** [comment]
Yeah the setting currently doesn’t prevent a focus ring on the tabs. I think because in that case the focus is given to the selected tab programmatically, since only one tab is actually focusable at any given time.
